GTG/plugins/gamify/gamify.py

```diff
@@ -0,0 +1,361 @@
+        self.headerbar_button = self.builder.get_object('gamify-headerbar')
+
+        self.headerbar = self.plugin_api.get_header()
+        self.headerbar.add(self.headerbar_button)
+
+    def remove_headerbar_button(self):
+        self.headerbar.remove(self.headerbar_button)
+
+    def add_levelbar(self):
+        self.quickadd_pane = self.plugin_api.get_quickadd_pane()
+        self.levelbar = self.builder.get_object('goal-levelbar')
+        self.quickadd_pane.set_orientation(Gtk.Orientation.VERTICAL)
+        self.quickadd_pane.add(self.levelbar)
+
+    def remove_levelbar(self):
+        self.quickadd_pane.set_orientation(Gtk.Orientation.HORIZONTAL)
+        self.quickadd_pane.remove(self.levelbar)
+
+    def activate(self, plugin_api):
+        self.plugin_api = plugin_api
+        self.browser = plugin_api.get_browser()
+
+        # Load preferences and data
+        self.analytics_load()
+        self.preferences_load()
+
+        # Settings up the menu 
+        self.add_ui()
+
+        # Init the preference dialog
+        try:
+            self._init_dialog_pref()
+        except ValueError:
+            self.configureable = False
+            log.debug('Cannot load preference dialog widget')
+
+        # Connect to the signals
+        self.connect_id = self.browser.connect("task-marked-as-done", self.on_marked_as_done)
+
+        self.update_date()
+        self.update_streak()
+        self.analytics_save()
+        self.update_widget()
+
+    def on_marked_as_done(self, sender, task_id):
+        log.debug('a task has been marked as done')
+        self.analytics_load()
+        self.preferences_load()
+
+        # Update the date, if it is different from today
+        self.update_date()
+
+        # Increase the number of tasks done and update the streak
+        # if the goal number of tasks was achieved
+        self.data['last_task_number'] += 1
+        self.update_streak()
+        self.data['score'] += self.get_points_for_task(task_id)
+        self.analytics_save()
+        self.update_widget()
+
+    def get_points_for_task(self, task_id):
+        """Returns the number of point for doing a perticular task
+
+        If a task is taged as @hard: the user receives 3 points
+        If a task is taged as @meduim: the use receives 2 points
+        If a task is taged as @easy: the user receives 1 point
+        """
+        easy = False
+        meduim = False
+        hard = False
+        task = self.plugin_api.get_requester().get_task(task_id)
+        tags = task.get_tags_name()
+        for tag in tags:
+            if tag == '@easy':
+                easy = True
+            elif tag == '@meduim':
+                meduim = True
+            elif tag == '@hard':
+                hard = True
+
+        if easy:
+            return 1
+        if meduim:
+            return 2
+        if hard:
+            return 3
+        return 1
+
+    def OnTaskOpened(self, plugin_api):
+        pass
```

Comment:
Did you want to do something here and changed your mind, or is the plugin API forcing this method to be present?